packages/e3kit-base/src/AbstractEThree.ts

```diff
@@ -0,0 +1,725 @@
+        if (!isValidParticipantCount(participantIdentities.size)) {
+            throw new GroupError(
+                GroupErrorCode.InvalidParticipantsCount,
+                `Cannot create group with ${participantIdentities.size} participant(s). Group can have ${VALID_GROUP_PARTICIPANT_COUNT_RANGE[0]} to ${VALID_GROUP_PARTICIPANT_COUNT_RANGE[1]} participants.`,
+            );
+        }
+
+        const groupSession = this.virgilCrypto.generateGroupSession(groupId);
+        const ticket = {
+            groupSessionMessage: {
+                epochNumber: groupSession.getCurrentEpochNumber(),
+                sessionId: groupSession.getSessionId(),
+                data: groupSession.export()[0].toString('base64'),
+            },
+            participants: [...participantIdentities],
+        };
+        return await this.groupManager.store(ticket, participantCards);
+    }
+
+    async loadGroup(groupId: Data, initiatorCard: ICard) {
+        const sessionId = this.virgilCrypto.calculateGroupSessionId(groupId);
+        return await this.groupManager.pull(sessionId, initiatorCard);
+    }
+
+    async getGroup(groupId: Data) {
+        const sessionId = this.virgilCrypto.calculateGroupSessionId(groupId);
+        return await this.groupManager.retrieve(sessionId);
+    }
+
+    async deleteGroup(groupId: Data) {
+        const sessionId = this.virgilCrypto.calculateGroupSessionId(groupId);
+        const group = await this.groupManager.retrieve(sessionId);
+        if (!group) {
+            throw new GroupError(
+                GroupErrorCode.LocalGroupNotFound,
+                `Group with ID "${groupId}" was not found in local storage. Try to load it first.`,
+            );
+        }
+        if (!group.isEditable()) {
+            throw new GroupError(
+                GroupErrorCode.PermissionDenied,
+                'Only group initiator can delete the group',
+            );
+        }
+
+        await this.groupManager.delete(sessionId);
+    }
+
+    /**
+     * @hidden
+     */
+    private async publishCard(
+        keyPair: IKeyPair,
+        previousCardId?: string,
+    ): Promise<{ keyPair: IKeyPair; card: ICard }> {
+        const card = await this.cardManager.publishCard({
+            privateKey: keyPair.privateKey,
+            publicKey: keyPair.publicKey,
+            previousCardId,
+        });
+
+        return { keyPair, card };
+    }
+
+    /**
+     * @hidden
+     */
+    private isOwnPublicKeyIncluded(ownPublicKey: IPublicKey, publicKeys: IPublicKey[]) {
+        const selfPublicKey = this.virgilCrypto.exportPublicKey(ownPublicKey).toString('base64');
+
+        const stringKeys = publicKeys.map(key =>
+            this.virgilCrypto.exportPublicKey(key).toString('base64'),
+        );
+        return stringKeys.some(key => key === selfPublicKey);
+    }
+
+    private throwIllegalInvocationError(method: string) {
+        throw new Error(`Calling ${method} two or more times in a row is not allowed.`);
```

Comment:
Afaik this method exists since early versions of e3kit. We use it in crucial methods that can be called multiple times by user without knowing anything about their asynchronous behaviour.
I'm also not against removing it :)
@xlwknx @vadimavdeev wdyt?

Reply:
This code solves next problem:
1. User send 2 register request in same time (possibly by developer mistake) 
2. Register will be successful and creates two cards with same identity
3. This identity is unusable in e3kit now, cause we have 1 card restrictions.
So my point: it is okay to replace this code with some better solution for this problem. But not just remove it, because it is not obvious why we let register 2 cards in the same time and then this is identity is cannot be used by `findUsers` and `lookupPublicKeys`.